script.js

```diff
@@ -1,4 +1,709 @@
+//         );
+//       }
+//       // Reset secondary arrays. Do not reset core array "competingNumbers"
+//       competingArrays = [];
+//       competingTotals = [];
+//       indivArray = [];
+//       indivTotal = [];
+//     }
+//   }
+// };
+
+// ======================= MORE COMFORTABLE (SCORE + LEADERBOARD + LOWEST COMBINED NUMBER MODE + AUTO-GENERATE COMBINED NUMBER + VARIABLE NUMBER OF DICE + VARIABLE NUMBER OF PLAYERS + KNOCKOUT MODE) ===========================
+
+// Create scream codes for frequently used strings
+var NORMAL = "normal";
+var REVERSED = "reversed";
+var NO_OF_PLAYERS = "enter number of players";
+var GAME = "game";
+var UNASSIGNED = "unassigned";
+
+// Create an array to store the newly rolled dice numbers for the current player
+var diceRollArray = [];
+// Create result mode with default being normal (i.e. highest score wins)
+var resultMode = NORMAL;
+// Create game mode where default is to enter number of players first
+var gameMode = NO_OF_PLAYERS;
+// Create counter to track which player's turn it is, where 'player 1' is 0, 'player 2' is 1 and so on..
+var playerNo = 0;
+// Create array to store all players' competing numbers in sequence of turn taken (e.g. player 1's, player 2's, .. player X's, player 1's, player 2's, .. player X's)
+var competingNumbers = [];
+// Create array to store all individual players' array of best numbers across previous and current rounds (e.g. player 1's array of numbers, player 2's array of numbers...)
+var competingArrays = [];
+// Create array to store the total of each player's best combined numbers (e.g. player 1's total, player 2's total...)
+var competingTotals = [];
+
+// Create global variable to store the number of dice selected at each round, default being "unassigned"
+var numberOfDice = UNASSIGNED;
+// Create global variable to store the number of players indicated at the start of the game
+var numberOfPlayers = ``;
+
+// Create function to check input is a number > 0
+var checkNumber = function (data) {
+  return !isNaN(Number(data)) && Number(data) != 0;
+};
+
+// Create a function for random dice roll
+var diceRoll = function () {
+  return Math.floor(Math.random() * 6) + 1;
+};
+// Create a function to roll a specific quantity of dice stated
+var rollAnyQtyDice = function (quantity) {
+  diceRollArray = [];
+  var counter = 0;
+  while (counter < quantity) {
+    diceRollArray.push(diceRoll());
+    counter += 1;
+  }
+  return diceRollArray;
+};
+
+// Sort an array in an ascending order
+var sortAsc = function (array) {
+  return array.sort(function (a, b) {
+    return a - b;
+  });
+};
+// Sort an array in a descending order
+var sortDes = function (array) {
+  return array.sort(function (a, b) {
+    return b - a;
+  });
+};
+
+// Store the unsequenced dice numbers rolled in the current round for the current player
+var storeRandomArray = function (array) {
+  var diceCounter = 0;
+  var unsortedarray = ``;
```

Comment:
For most languages, JavaScript included, although you have template literals ````
It will be good to declare: 
`var unsortedarray = "" `
or 
`var unsortedarray = '';`
so if you do switch languages, won't have to rethink so much